src/main/java/org/zalando/nakadi/security/ScopeHelper.java

```diff
@@ -0,0 +1,25 @@
+package org.zalando.nakadi.security;
+
+import org.zalando.nakadi.exceptions.IllegalScopeException;
+
+import java.util.Objects;
+import java.util.Set;
+
+public class ScopeHelper {
+
+    public static void checkScopes(final Set<String> allowedScopes, final Set<String> clientScopes) {
+        if (isEmptyScope(allowedScopes) ) {
+            return;
+        }
+
+        allowedScopes.stream()
+                .filter(clientScopes::contains)
+                .findAny()
+                .orElseThrow(() -> new IllegalScopeException(allowedScopes));
+    }
+
+    private static boolean isEmptyScope(final Set<String> allowedScopes) {
+        return Objects.isNull(allowedScopes) || allowedScopes.isEmpty();
```

Comment:
1) Why you check for null in such a strange way? :)
2) Why allowedScopes can be null? Our null handling strategy supposes that everything is none-null by default, and if something can be null is should be marked as `@Nullable`. But I really don't see a reason for allowedScopes to be null, I think in EventType class you should actually set it to empty set if the null is passed in setter.